pipeline/scrape_live_test.py

```diff
@@ -0,0 +1,98 @@
+"""Scraping a live game with the Sportsmonks API."""
+
+from os import environ as ENV, path
+from datetime import datetime, timezone
+import json
+import time
+import http.client
+
+from dotenv import load_dotenv
+
+
+def scrape_live_match(match_identifier: str | int,
+                      token: str, conn: http.client.HTTPSConnection) -> dict:
+    """Returns a dict of scraped information for a specified match."""
+
+    payload = ''
+    headers = {}
+
+    if isinstance(match_identifier, str):
+        url = f"livescores/inplay?api_token={token}&include=statistics&filters=participantSearch:{match_identifier}"
+    elif isinstance(match_identifier, int):
+        url = f"fixtures/{match_identifier}?api_token={token}"
+    else:
+        raise TypeError(
+            f"{match_identifier} is not a valid string or integer.")
+
+    conn.request(
+        "GET", f"/v3/football/{url}", payload, headers)
+    res = conn.getresponse()
+    data = res.read()
+    data_str = data.decode("utf-8")
+
+    return json.loads(data_str)
+
+
+def write_to_file(filename: str, data: dict) -> None:
+    """Appends scraped data to a json file."""
+
+    data = prepare_data(data)
+
+    if not path.exists(filename) or path.getsize(filename) == 0:
+        with open(filename, 'w', encoding='utf-8') as f:
+            json.dump([data], f, indent=4)
+    else:
+
+        with open(filename, "r+", encoding="utf-8") as f:
+            file_data = json.load(f)
+            file_data.append(data)
+            f.seek(0)
+            json.dump(file_data, f, indent=4)
+
+
+def prepare_data(data: dict) -> dict:
+    """
+    Removes unwanted keys from our api call and adds info.
+    NOTE: Whilst this is transformative, we need to capture a timestamp of when it was
+    extracted at the time of extraction, and we don't want to write unnecessary keys to 
+    our file each time. I've kept this minimal.
+    """
+    for dict_key in ["subscription", "timezone"]:
```

Comment:
```suggestion
    for dict_key in {"subscription", "timezone"}:
```

For minor efficiency gains

Reply:
I think you mean a tuple, as opposed to `{}`. I'll add it!

Reply:
@AhmetCengiz03 You might be right, I found this [old stackoverflow thread](https://stackoverflow.com/questions/25272400/which-one-is-faster-iterating-over-a-set-and-iterating-over-a-list) on it though that makes me err on the side of the tuple!

Reply:
Yeah I'm happy with either, I think there's such a minimal difference in efficiency that it will be fine